oshi-core/src/main/java/oshi/util/platform/windows/RegistryUtil.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2025 The OSHI Project Contributors
+ * SPDX-License-Identifier: MIT
+ */
+package oshi.util.platform.windows;
+
+import com.sun.jna.platform.win32.Advapi32;
+import com.sun.jna.platform.win32.Advapi32Util;
+import com.sun.jna.platform.win32.Win32Exception;
+import com.sun.jna.platform.win32.WinReg;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import oshi.util.ParseUtil;
+
+import com.sun.jna.platform.win32.WinReg.HKEY;
+import com.sun.jna.platform.win32.WinReg.HKEYByReference;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Locale;
+
+import static com.sun.jna.platform.win32.WinError.ERROR_SUCCESS;
+import static com.sun.jna.platform.win32.WinNT.KEY_READ;
+
+public final class RegistryUtil {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RegistryUtil.class);
+
+    private static final long THIRTY_YEARS_IN_SECS = 30L * 365 * 24 * 60 * 60;
+
+    private static final Advapi32 ADV = Advapi32.INSTANCE;
+
+    private RegistryUtil() {
+    }
+
+    /**
+     * Returns a registry value as a Long. (without access flag) Currently supports String and Integer
+     */
+    public static long getLongValue(HKEY root, String path, String key) {
+        try {
+            Object val = Advapi32Util.registryGetValue(root, path, key);
+            return registryValueToLong(val);
+        } catch (Win32Exception e) {
+            LOG.trace("Unable to access " + path + ": " + e.getMessage());
+        }
+        return 0L;
+    }
+
+    /**
+     * Returns a registry value as a Long. (with access flag) Currently supports String and Integer
+     */
+    public static long getLongValue(HKEY root, String path, String key, int accessFlag) {
+        Object val = getRegistryValueOrNull(root, path, key, accessFlag);
+        return registryValueToLong(val);
+    }
+
+    public static Object getRegistryValueOrNull(HKEY root, String path, String key, int accessFlag) {
+        HKEY hKey = null;
+        try {
+            hKey = getRegistryHKey(root, path, accessFlag);
+            Object value = Advapi32Util.registryGetValue(root, path, key);
+            if ((value instanceof Integer) || (value instanceof String && !((String) value).trim().isEmpty())
+                    || value instanceof byte[]) {
+                return value;
+            }
```

Comment:
```suggestion
            return Objects.isNull(value) ? null : value;
```
In addition to removing the unneeded `getRegistryHkey()` and the whole finally block.

Reply:
Ignore my comment about the finally block: I was trusting the copilot auto-comment (who turned that on, anyway? 😁).   It's needed.  Otherwise looks good.